WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift

```diff
@@ -1255,6 +1266,42 @@ extension EditableOrderViewModel {
     func requestCameraAccess(onCompletion: @escaping ((Bool) -> Void)) {
         permissionChecker.requestAccess(for: .video, completionHandler: onCompletion)
     }
+
+    /// Attempts to add a Product to the current Order by SKU search
+    ///
+    func addScannedProductToOrder(barcode sku: String?, onCompletion: @escaping (Result<Void, Error>) -> Void) {
+        guard let sku = sku else {
+            return onCompletion(.failure(ScannerError.nilSKU))
+        }
+
+        mapFromSKUtoProductID(sku: sku) { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case let .success(productID):
+                self.configureProductRowViewModels()
+                self.orderSynchronizer.setProduct.send(.init(product: .productID(productID), quantity: 1))
```

Comment:
Yes, I initially planned to add a check to see if the product already exist in the order, and increase its quantity if it does, rather than sending it again to the synchronizer, however as we're gravitating to use this feature only when creating new orders for now ( pecCkj-F1#comment-308-p2 ) I left it as it is for now as the quantity will always be 1.